### PR TITLE
bpf: Add send_trace_notify hook for redirect_direct_{v4,v6}

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg.

### CODE REVIEW

**Comment by borkmann** [comment]
/test-1.23-net-next

File: bpf/bpf_lxc.c [hunk 1]
```diff
@@ -937,8 +945,16 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
...
 	}
 #endif /* TUNNEL_MODE */
-	if (is_defined(ENABLE_HOST_ROUTING))
-		return redirect_direct_v4(ctx, l3_off, ip4);
+	if (is_defined(ENABLE_HOST_ROUTING)) {
+		int oif;
+
+		ret = redirect_direct_v4(ctx, l3_off, ip4, &oif);
+		if (likely(ret == CTX_ACT_REDIRECT))
+			send_trace_notify(ctx, TRACE_TO_NETWORK, SECLABEL,
```
Comment (brb) [inline bpf/bpf_lxc.c:953]: Did the problem of the missing trace appeared when bpf_lxc was doing the redirect to a netdev which didn't have bpf_host attached to the TC egress? Asking, as the `to-netdev` section of bpf_host don't send the trace due to the bug described in https://github.com/cilium/cilium/issues/18909.

**Review by pchaigno — APPROVED** [review]

**Comment by borkmann** [comment]
Closing given https://github.com/cilium/cilium/pull/19158 was merged which fixes this one too.

**Pull request closed without merging**